### DISCUSSION TITLE
How to install specific version of Storybook?

### DISCUSSION THREAD

**Discussion by Cloud9Clone** [storybookjs/storybook#23107] [Help]
I know that the question might be trivial but I didn't find any other source so I am sorry if this is already answered. 

We have an Angular project running version 13 and we would like to introduce the best possible version candidate of Storybook. As of my understanding, the project would be compatible with Storybook 6.5. However, when I attempt to run the following command `npx storybook@6.5.16 init` I can see that still version `7.0.21` gets installed. 

How can we install the specific version with `npx sb init`?

**Comment by f3bw** [comment]
To install a specific version of Storybook using `npx sb init`, you can use the `--template` flag followed by the version number you want. Here's the correct command to install Storybook version 6.5.16:

```bash
npx sb@6.5.16 init
```

By specifying `sb@6.5.16`, you are telling `npx` to install and run Storybook version 6.5.16.

If you continue to encounter issues with the installation, you can try clearing your npm cache by running `npm cache clean --force` before attempting the installation again.

**Reply by YasielCabrera** [reply]
I have tried both, `sb` and `storybook` and both of them install the latest version. Only those were installed with the correct version

```
"@storybook/builder-webpack4": "^6.5.16",
"@storybook/manager-webpack4": "^6.5.16",
```

**Comment by cefn** [comment]
If you have a known working version of Storybook is there any way to install it?

**Reply by Marklb** [reply]
I am not positive if there is a specific version that it will work back to, but based on the docs example, it seems v7.6+, at least, should support installing specific versions.
https://storybook.js.org/docs/get-started/install

> The init command will use whichever version you specify. For example:
>
> - storybook@latest init will initialize the latest version
> - storybook@7.6.10 init will initialize 7.6.10
> - storybook@7 init will initialize the newest 7.x.x version

**Comment by cefn** [comment]
Thanks for the helpful response! 

It does seem since 7.6 the behaviour is correct. Unfortunately I had aligned with an earlier version of 7 (as I was migrating the storybook bundler from webpack to vite and wanted to introduce a minimum of compatibility issues). Rolling forward to the latest 7 is actually fine. I just needed to avoid rolling to version 8 and picking up all the unnecessary migration issues.

For my case in node 20, creating a new package.json, installing typescript, running `npm run env -- tsc --init` and then `npx storybook@7.6.20 init --builder vite` got me an equivalent clean typescript storybook build with the vite bundler into which I could copy the project-specific story resources to complete the migration.
